addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -404,10 +406,18 @@ export class Orderline extends PosModel {
         this.price_type = json.price_type;
         this.set_discount(json.discount);
         this.set_quantity(json.qty, "do not recompute unit price");
-        this.set_description(json.description);
         this.attribute_value_ids = json.attribute_value_ids || [];
         this.set_price_extra(json.price_extra);
-        this.set_full_product_name(json.full_product_name);
+        this.custom_attribute_value_ids =
+            json.custom_attribute_value_ids.reduce((acc, attr) => {
+                if (attr.length > 0) {
+                    attr = attr[2];
+                }
+
+                acc.push(new ProductCustomAttribute(attr));
+                return acc;
+            }, []) || [];
```

Comment:
```suggestion
        this.custom_attribute_value_ids =
            json.custom_attribute_value_ids.map((attr) => {
                if (attr.length > 0) {
                    attr = attr[2];
                }
                return new ProductCustomAttribute(attr);
            });
```

Change "reduce" to a "map".